packages/typespec-python/src/http.ts

```diff
@@ -331,7 +331,12 @@ function emitHttpResponse(
     let type = undefined;
     if (response.type && !isAzureCoreModel(response.type)) {
         type = getType(context, response.type);
-    } else if (method && method.response.type && !isAzureCoreModel(method.response.type)) {
+    } else if (
+        method &&
+        method.response.type &&
+        method.kind.includes("lro") &&
+        !isAzureCoreModel(method.response.type)
+    ) {
```

Comment:
For LRO API, there is always one kind of return type; but for non-lro API, there may be multi kinds of response type. Although we are not sure how to design for multi responses but current behavior is not right.